Makefile

```diff
@@ -286,8 +287,45 @@ compile_clarity:
 	fi
 	@echo "Done."
 
+compile_clarity_harborui:
+	@echo "compiling binary for clarity ui with local harbor ui library..."
+	@if [ ! -e $(UINGPATH)/package.json.org ] ; then \
+		cp $(UINGPATH)/package.json $(UINGPATH)/package.json.org; \
+		sed -i -e "s/\"harbor-ui\": \"\^[0-9].\+\"/\"harbor-ui\": \"file:.\/lib\/\"/g" $(UINGPATH)/package.json; \
+	fi
+	@if [ "$(HTTPPROXY)" != "" ] ; then \
+		$(SEDCMD) -i 's/__proxy__/--proxy $(HTTPPROXY)/g' $(DOCKERFILE_BUILD_UI) ; \
+		$(DOCKERBUILD) -f $(DOCKERFILE_BUILD_UI) -t $(DOCKERIMAGENAME_CLATIRY):dev .; \
+		$(DOCKERCMD) run --rm -v $(BUILDPATH)/src:$(CLARITYSEEDPATH) $(DOCKERIMAGENAME_CLATIRY):dev $(SHELL) $(CLARITYBUILDSCRIPT) -p $(HTTPPROXY); \
+	else \
+		$(SEDCMD) -i 's/__proxy__/ /g' $(DOCKERFILE_BUILD_UI) ; \
+		$(DOCKERBUILD) -f $(DOCKERFILE_BUILD_UI) -t $(DOCKERIMAGENAME_CLATIRY):dev .; \
+		$(DOCKERCMD) run --rm -v $(BUILDPATH)/src:$(CLARITYSEEDPATH) $(DOCKERIMAGENAME_CLATIRY):dev $(SHELL) $(CLARITYBUILDSCRIPT); \
+	fi
+	@if [ -e $(UINGPATH)/package.json.org ] ; then \
+		mv $(UINGPATH)/package.json.org $(UINGPATH)/package.json; \
+	fi
+	@echo "Done."
+
 compile_normal: compile_clarity compile_adminserver compile_ui compile_jobservice
 
+compile_harborui: compile_clarity_harborui
```

Comment:
The target name seems not tell the exactly jobs here, please give an reasonable one.

Can you give an example on how to use this target in Make?  You can refer to this doc. https://github.com/vmware/harbor/blob/master/docs/use_make.md

Reply:
thank you for review.
I think this name is fit at jobs here.
compile_harborui -> compile_clarity_with_local_harborui
Is it OK?

Reply:
It's not just compile UI, but adminserver/jobservice. 

compile_golangimage_with_local_library, is it OK? Again, please commit example and description in the use_make.md.

Reply:
OK! 
thanks!